@daschuer I get a build error on this code:

```
src/preferences/dialog/dlgpreflibrary.cpp: In member function ‘virtual void DlgPrefLibrary::slotUpdate()’:
src/preferences/dialog/dlgpreflibrary.cpp:178:13: error: ‘radioButton_dbclick_edit_metadata’ was not declared in this scope
             radioButton_dbclick_edit_metadata->setChecked(true);
             ^
src/preferences/dialog/dlgpreflibrary.cpp: In member function ‘virtual void DlgPrefLibrary::slotApply()’:
src/preferences/dialog/dlgpreflibrary.cpp:309:16: error: ‘radioButton_dbclick_edit_metadata’ was not declared in this scope
     } else if (radioButton_dbclick_edit_metadata->isChecked()) {
```

--- a/src/library/autodj/dlgautodj.cpp
+++ b/src/library/autodj/dlgautodj.cpp
@@ -29,13 +29,15 @@ DlgAutoDJ::DlgAutoDJ(QWidget* parent,
             this, SIGNAL(loadTrackToPlayer(TrackPointer, QString, bool)));
     connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
             this, SIGNAL(trackSelected(TrackPointer)));
+    connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
+            this, SLOT(updateSelectionInfo()));
+
     connect(pLibrary, SIGNAL(setTrackTableFont(QFont)),
             m_pTrackTableView, SLOT(setTrackTableFont(QFont)));
     connect(pLibrary, SIGNAL(setTrackTableRowHeight(int)),
             m_pTrackTableView, SLOT(setTrackTableRowHeight(int)));
-    connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
-            this, SLOT(updateSelectionInfo()));
-
+    connect(pLibrary, SIGNAL(setSelectedClick(bool)),
+            m_pTrackTableView, SLOT(setSelectedClick(bool)));
 
     QBoxLayout* box = dynamic_cast<QBoxLayout*>(layout());
     VERIFY_OR_DEBUG_ASSERT(box) { //Assumes the form layout is a QVBox/QHBoxLayout!
@@ -189,14 +191,6 @@ void DlgAutoDJ::autoDJStateChanged(AutoDJProcessor::AutoDJState state) {
         // You can always skip the next track if we are enabled.
         pushButtonSkipNext->setEnabled(true);
     }
-}
-
-void DlgAutoDJ::setTrackTableFont(const QFont& font) {
-    m_pTrackTableView->setTrackTableFont(font);
-}
-
-void DlgAutoDJ::setTrackTableRowHeight(int rowHeight) {
-    m_pTrackTableView->setTrackTableRowHeight(rowHeight);
 }
 
 void DlgAutoDJ::updateSelectionInfo() {

--- a/src/library/autodj/dlgautodj.h
+++ b/src/library/autodj/dlgautodj.h
@@ -40,8 +40,6 @@ class DlgAutoDJ : public QWidget, public Ui::DlgAutoDJ, public LibraryView {
     void transitionTimeChanged(int time);
     void transitionSliderChanged(int value);
     void autoDJStateChanged(AutoDJProcessor::AutoDJState state);
-    void setTrackTableFont(const QFont& font);
-    void setTrackTableRowHeight(int rowHeight);
     void updateSelectionInfo();
 
   signals:

--- a/src/library/dlganalysis.cpp
+++ b/src/library/dlganalysis.cpp
@@ -8,10 +8,10 @@
 #include "library/dlganalysis.h"
 #include "util/assert.h"
 
-DlgAnalysis::DlgAnalysis(QWidget* parent,
+DlgAnalysis::DlgAnalysis(QWidget* pLibrary,
                        UserSettingsPointer pConfig,
                        TrackCollection* pTrackCollection)
-        : QWidget(parent),
+        : QWidget(pLibrary),
           m_pConfig(pConfig),
           m_pTrackCollection(pTrackCollection),
           m_bAnalysisActive(false),
@@ -62,6 +62,13 @@ DlgAnalysis::DlgAnalysis(QWidget* parent,
             SIGNAL(selectionChanged(const QItemSelection &, const QItemSelection&)),
             this,
             SLOT(tableSelectionChanged(const QItemSelection &, const QItemSelection&)));
+
+    connect(pLibrary, SIGNAL(setTrackTableFont(QFont)),
+            m_pAnalysisLibraryTableView, SLOT(setTrackTableFont(QFont)));
+    connect(pLibrary, SIGNAL(setTrackTableRowHeight(int)),
+            m_pAnalysisLibraryTableView, SLOT(setTrackTableRowHeight(int)));
+    connect(pLibrary, SIGNAL(setSelectedClick(bool)),
+            m_pAnalysisLibraryTableView, SLOT(setSelectedClick(bool)));
 }
 
 DlgAnalysis::~DlgAnalysis() {

--- a/src/library/dlghidden.cpp
+++ b/src/library/dlghidden.cpp
@@ -41,13 +41,15 @@ DlgHidden::DlgHidden(QWidget* parent, UserSettingsPointer pConfig,
             SIGNAL(selectionChanged(const QItemSelection&, const QItemSelection&)),
             this,
             SLOT(selectionChanged(const QItemSelection&, const QItemSelection&)));
-
     connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
             this, SIGNAL(trackSelected(TrackPointer)));
+
     connect(pLibrary, SIGNAL(setTrackTableFont(QFont)),
             m_pTrackTableView, SLOT(setTrackTableFont(QFont)));
     connect(pLibrary, SIGNAL(setTrackTableRowHeight(int)),
             m_pTrackTableView, SLOT(setTrackTableRowHeight(int)));
+    connect(pLibrary, SIGNAL(setSelectedClick(bool)),
+            m_pTrackTableView, SLOT(setSelectedClick(bool)));
 }
 
 DlgHidden::~DlgHidden() {
@@ -85,14 +87,6 @@ void DlgHidden::selectionChanged(const QItemSelection &selected,
                                  const QItemSelection &deselected) {
     Q_UNUSED(deselected);
     activateButtons(!selected.indexes().isEmpty());
-}
-
-void DlgHidden::setTrackTableFont(const QFont& font) {
-    m_pTrackTableView->setTrackTableFont(font);
-}
-
-void DlgHidden::setTrackTableRowHeight(int rowHeight) {
-    m_pTrackTableView->setTrackTableRowHeight(rowHeight);
 }
 
 bool DlgHidden::hasFocus() const {

--- a/src/library/dlghidden.h
+++ b/src/library/dlghidden.h
@@ -28,8 +28,6 @@ class DlgHidden : public QWidget, public Ui::DlgHidden, public LibraryView {
     void clicked();
     void selectAll();
     void selectionChanged(const QItemSelection&, const QItemSelection&);
-    void setTrackTableFont(const QFont& font);
-    void setTrackTableRowHeight(int rowHeight);
 
   signals:
     void trackSelected(TrackPointer pTrack);

--- a/src/library/dlgmissing.cpp
+++ b/src/library/dlgmissing.cpp
@@ -36,13 +36,15 @@ DlgMissing::DlgMissing(QWidget* parent, UserSettingsPointer pConfig,
             SIGNAL(selectionChanged(const QItemSelection&, const QItemSelection&)),
             this,
             SLOT(selectionChanged(const QItemSelection&, const QItemSelection&)));
+    connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
+            this, SIGNAL(trackSelected(TrackPointer)));
+
     connect(pLibrary, SIGNAL(setTrackTableFont(QFont)),
             m_pTrackTableView, SLOT(setTrackTableFont(QFont)));
     connect(pLibrary, SIGNAL(setTrackTableRowHeight(int)),
             m_pTrackTableView, SLOT(setTrackTableRowHeight(int)));
-
-    connect(m_pTrackTableView, SIGNAL(trackSelected(TrackPointer)),
-            this, SIGNAL(trackSelected(TrackPointer)));
+    connect(pLibrary, SIGNAL(setSelectedClick(bool)),
+            m_pTrackTableView, SLOT(setSelectedClick(bool)));
 }
 
 DlgMissing::~DlgMissing() {
@@ -78,14 +80,6 @@ void DlgMissing::selectionChanged(const QItemSelection &selected,
                                   const QItemSelection &deselected) {
     Q_UNUSED(deselected);
     activateButtons(!selected.indexes().isEmpty());
-}
-
-void DlgMissing::setTrackTableFont(const QFont& font) {
-    m_pTrackTableView->setTrackTableFont(font);
-}
-
-void DlgMissing::setTrackTableRowHeight(int rowHeight) {
-    m_pTrackTableView->setTrackTableRowHeight(rowHeight);
 }
 
 bool DlgMissing::hasFocus() const {

--- a/src/library/dlgmissing.h
+++ b/src/library/dlgmissing.h
@@ -27,8 +27,6 @@ class DlgMissing : public QWidget, public Ui::DlgMissing, public LibraryView {
     void clicked();
     void selectAll();
     void selectionChanged(const QItemSelection&, const QItemSelection&);
-    void setTrackTableFont(const QFont& font);
-    void setTrackTableRowHeight(int rowHeight);
 
   signals:
     void trackSelected(TrackPointer pTrack);

--- a/src/library/library.cpp
+++ b/src/library/library.cpp
@@ -172,6 +172,9 @@ Library::Library(
     } else {
         m_trackTableFont = QApplication::font();
     }
+
+    m_editMetadata = m_pConfig->getValue(
+            ConfigKey(kConfigGroup, "EditMetadata"), true);
 }
 
 Library::~Library() {
@@ -242,6 +245,8 @@ void Library::bindWidget(WLibrary* pLibraryWidget,
             pTrackTableView, SLOT(setTrackTableFont(QFont)));
     connect(this, SIGNAL(setTrackTableRowHeight(int)),
             pTrackTableView, SLOT(setTrackTableRowHeight(int)));
+    connect(this, SIGNAL(setSelectedClick(bool)),
+            pTrackTableView, SLOT(setSelectedClick(bool)));
 
     connect(this, SIGNAL(searchStarting()),
             pTrackTableView, SLOT(onSearchStarting()));
@@ -260,6 +265,7 @@ void Library::bindWidget(WLibrary* pLibraryWidget,
     // just connected to us.
     emit(setTrackTableFont(m_trackTableFont));
     emit(setTrackTableRowHeight(m_iTrackTableRowHeight));
+    emit(setSelectedClick(m_editMetadata));
 }
 
 void Library::addFeature(LibraryFeature* feature) {
@@ -412,14 +418,19 @@ QStringList Library::getDirs() {
     return m_pTrackCollection->getDirectoryDAO().getDirs();
 }
 
-void Library::slotSetTrackTableFont(const QFont& font) {
+void Library::setFont(const QFont& font) {
     m_trackTableFont = font;
     emit(setTrackTableFont(font));
 }
 
-void Library::slotSetTrackTableRowHeight(int rowHeight) {
+void Library::setRowHeight(int rowHeight) {
     m_iTrackTableRowHeight = rowHeight;
     emit(setTrackTableRowHeight(rowHeight));
+}
+
+void Library::setEditMedatata(bool enabled) {
+    m_editMetadata = enabled;
+    emit(setSelectedClick(enabled));
 }
 
 void Library::saveCachedTrack(Track* pTrack) noexcept {

--- a/src/library/library.cpp
+++ b/src/library/library.cpp
@@ -173,8 +173,8 @@ Library::Library(
         m_trackTableFont = QApplication::font();
     }
 
-    m_editMetadata = m_pConfig->getValue(
-            ConfigKey(kConfigGroup, "EditMetadata"), true);
+    m_editMetadataSelectedClick = m_pConfig->getValue(
+            ConfigKey(kConfigGroup, "EditMetadataSelectedClick"), true);
 }
 
 Library::~Library() {
@@ -265,7 +265,7 @@ void Library::bindWidget(WLibrary* pLibraryWidget,
     // just connected to us.
     emit(setTrackTableFont(m_trackTableFont));
     emit(setTrackTableRowHeight(m_iTrackTableRowHeight));
-    emit(setSelectedClick(m_editMetadata));
+    emit(setSelectedClick(m_editMetadataSelectedClick));
 }
 
 void Library::addFeature(LibraryFeature* feature) {
@@ -428,8 +428,8 @@ void Library::setRowHeight(int rowHeight) {
     emit(setTrackTableRowHeight(rowHeight));
 }
 
-void Library::setEditMedatata(bool enabled) {
-    m_editMetadata = enabled;
+void Library::setEditMedatataSelectedClick(bool enabled) {
+    m_editMetadataSelectedClick = enabled;
     emit(setSelectedClick(enabled));
 }
 

--- a/src/library/library.h
+++ b/src/library/library.h
@@ -81,6 +81,10 @@ class Library: public QObject,
 
     static const int kDefaultRowHeightPx;
 
+    void setFont(const QFont& font);
+    void setRowHeight(int rowHeight);
+    void setEditMedatata(bool enable);
+
   public slots:
     void slotShowTrackModel(QAbstractItemModel* model);
     void slotSwitchToView(const QString& view);
@@ -95,8 +99,6 @@ class Library: public QObject,
     void slotRequestRemoveDir(QString directory, Library::RemovalType removalType);
     void slotRequestRelocateDir(QString previousDirectory, QString newDirectory);
     void onSkinLoadFinished();
-    void slotSetTrackTableFont(const QFont& font);
-    void slotSetTrackTableRowHeight(int rowHeight);
 
     void scan() {
         m_scanner.scan();
@@ -117,6 +119,7 @@ class Library: public QObject,
 
     void setTrackTableFont(const QFont& font);
     void setTrackTableRowHeight(int rowHeight);
+    void setSelectedClick(bool enable);
 
     // Emitted when a library scan starts and finishes.
     void scanStarted();
@@ -144,6 +147,7 @@ class Library: public QObject,
     LibraryScanner m_scanner;
     QFont m_trackTableFont;
     int m_iTrackTableRowHeight;
+    bool m_editMetadata;
     QScopedPointer<ControlObject> m_pKeyNotation;
 };
 

--- a/src/library/library.h
+++ b/src/library/library.h
@@ -83,7 +83,7 @@ class Library: public QObject,
 
     void setFont(const QFont& font);
     void setRowHeight(int rowHeight);
-    void setEditMedatata(bool enable);
+    void setEditMedatataSelectedClick(bool enable);
 
   public slots:
     void slotShowTrackModel(QAbstractItemModel* model);
@@ -147,7 +147,7 @@ class Library: public QObject,
     LibraryScanner m_scanner;
     QFont m_trackTableFont;
     int m_iTrackTableRowHeight;
-    bool m_editMetadata;
+    bool m_editMetadataSelectedClick;
     QScopedPointer<ControlObject> m_pKeyNotation;
 };
 

--- a/src/library/librarycontrol.cpp
+++ b/src/library/librarycontrol.cpp
@@ -471,7 +471,7 @@ void LibraryControl::slotFontSize(double v) {
     }
     QFont font = m_pLibrary->getTrackTableFont();
     font.setPointSizeF(font.pointSizeF() + v);
-    m_pLibrary->slotSetTrackTableFont(font);
+    m_pLibrary->setFont(font);
 }
 
 void LibraryControl::slotIncrementFontSize(double v) {

--- a/src/library/recording/dlgrecording.cpp
+++ b/src/library/recording/dlgrecording.cpp
@@ -31,6 +31,8 @@ DlgRecording::DlgRecording(QWidget* parent, UserSettingsPointer pConfig,
             m_pTrackTableView, SLOT(setTrackTableFont(QFont)));
     connect(pLibrary, SIGNAL(setTrackTableRowHeight(int)),
             m_pTrackTableView, SLOT(setTrackTableRowHeight(int)));
+    connect(pLibrary, SIGNAL(setSelectedClick(bool)),
+            m_pTrackTableView, SLOT(setSelectedClick(bool)));
 
     connect(m_pRecordingManager, SIGNAL(isRecording(bool)),
             this, SLOT(slotRecordingEnabled(bool)));
@@ -157,11 +159,3 @@ void DlgRecording::refreshLabel() {
               .arg(m_durationRecordedStr);
     label->setText(text);
  }
-
-void DlgRecording::setTrackTableFont(const QFont& font) {
-    m_pTrackTableView->setTrackTableFont(font);
-}
-
-void DlgRecording::setTrackTableRowHeight(int rowHeight) {
-    m_pTrackTableView->setTrackTableRowHeight(rowHeight);
-}

--- a/src/library/recording/dlgrecording.h
+++ b/src/library/recording/dlgrecording.h
@@ -42,8 +42,6 @@ class DlgRecording : public QWidget, public Ui::DlgRecording, public virtual Lib
     void refreshBrowseModel();
     void slotRestoreSearch();
     void slotDurationRecorded(QString durationRecorded);
-    void setTrackTableFont(const QFont& font);
-    void setTrackTableRowHeight(int rowHeight);
 
   signals:
     void loadTrack(TrackPointer tio);

--- a/src/preferences/dialog/dlgpreflibrary.cpp
+++ b/src/preferences/dialog/dlgpreflibrary.cpp
@@ -139,7 +139,7 @@ void DlgPrefLibrary::slotResetToDefaults() {
     checkBox_show_itunes->setChecked(true);
     checkBox_show_traktor->setChecked(true);
     radioButton_dbclick_bottom->setChecked(false);
-    checkBoxEditMetadata->setChecked(true);
+    checkBoxEditMetadataSelectedClicked->setChecked(true);
     radioButton_dbclick_top->setChecked(false);
     radioButton_dbclick_deck->setChecked(true);
     spinBoxRowHeight->setValue(Library::kDefaultRowHeightPx);
@@ -176,9 +176,9 @@ void DlgPrefLibrary::slotUpdate() {
             break;
     }
 
-    checkBoxEditMetadata->setChecked(
+    checkBoxEditMetadataSelectedClicked->setChecked(
             m_pConfig->getValue(
-                        ConfigKey("[Library]","EditMetadata"), true));
+                        ConfigKey("[Library]","EditMetadataSelectedClick"), true));
 
     m_originalTrackTableFont = m_pLibrary->getTrackTableFont();
     m_iOriginalTrackTableRowHeight = m_pLibrary->getTrackTableRowHeight();
@@ -310,9 +310,10 @@ void DlgPrefLibrary::slotApply() {
     m_pConfig->set(ConfigKey("[Library]","TrackLoadAction"),
                 ConfigValue(dbclick_status));
 
-    m_pConfig->set(ConfigKey("[Library]","EditMetadata"),
-                    ConfigValue(checkBoxEditMetadata->checkState()));
-    m_pLibrary->setEditMedatata(checkBoxEditMetadata->checkState());
+    m_pConfig->set(ConfigKey("[Library]", "EditMetadataSelectedClick"),
+            ConfigValue(checkBoxEditMetadataSelectedClicked->checkState()));
+    m_pLibrary->setEditMedatataSelectedClick(
+            checkBoxEditMetadataSelectedClicked->checkState());
 
     QFont font = m_pLibrary->getTrackTableFont();
     if (m_originalTrackTableFont != font) {

--- a/src/preferences/dialog/dlgpreflibrary.cpp
+++ b/src/preferences/dialog/dlgpreflibrary.cpp
@@ -57,10 +57,6 @@ DlgPrefLibrary::DlgPrefLibrary(
 
     connect(libraryFontButton, SIGNAL(clicked()),
             this, SLOT(slotSelectFont()));
-    connect(this, SIGNAL(setTrackTableFont(QFont)),
-            m_pLibrary, SLOT(slotSetTrackTableFont(QFont)));
-    connect(this, SIGNAL(setTrackTableRowHeight(int)),
-            m_pLibrary, SLOT(slotSetTrackTableRowHeight(int)));
 
     // TODO(XXX) this string should be extracted from the soundsources
     QString builtInFormatsStr = "Ogg Vorbis, FLAC, WAVe, AIFF";
@@ -143,6 +139,7 @@ void DlgPrefLibrary::slotResetToDefaults() {
     checkBox_show_itunes->setChecked(true);
     checkBox_show_traktor->setChecked(true);
     radioButton_dbclick_bottom->setChecked(false);
+    checkBoxEditMetadata->setChecked(true);
     radioButton_dbclick_top->setChecked(false);
     radioButton_dbclick_deck->setChecked(true);
     spinBoxRowHeight->setValue(Library::kDefaultRowHeightPx);
@@ -174,13 +171,14 @@ void DlgPrefLibrary::slotUpdate() {
     case ADD_TO_AUTODJ_TOP:
             radioButton_dbclick_top->setChecked(true);
             break;
-    case EDIT_METADATA:
-            radioButton_dbclick_edit_metadata->setChecked(true);
-            break;
     default:
             radioButton_dbclick_deck->setChecked(true);
             break;
     }
+
+    checkBoxEditMetadata->setChecked(
+            m_pConfig->getValue(
+                        ConfigKey("[Library]","EditMetadata"), true));
 
     m_originalTrackTableFont = m_pLibrary->getTrackTableFont();
     m_iOriginalTrackTableRowHeight = m_pLibrary->getTrackTableRowHeight();
@@ -190,8 +188,8 @@ void DlgPrefLibrary::slotUpdate() {
 
 void DlgPrefLibrary::slotCancel() {
     // Undo any changes in the library font or row height.
-    emit(setTrackTableRowHeight(m_iOriginalTrackTableRowHeight));
-    emit(setTrackTableFont(m_originalTrackTableFont));
+    m_pLibrary->setFont(m_originalTrackTableFont);
+    m_pLibrary->setRowHeight(m_iOriginalTrackTableRowHeight);
 }
 
 void DlgPrefLibrary::slotAddDir() {
@@ -306,13 +304,15 @@ void DlgPrefLibrary::slotApply() {
             dbclick_status = ADD_TO_AUTODJ_BOTTOM;
     } else if (radioButton_dbclick_top->isChecked()) {
             dbclick_status = ADD_TO_AUTODJ_TOP;
-    } else if (radioButton_dbclick_edit_metadata->isChecked()) {
-            dbclick_status = EDIT_METADATA;
     } else {
             dbclick_status = LOAD_TO_DECK;
     }
     m_pConfig->set(ConfigKey("[Library]","TrackLoadAction"),
                 ConfigValue(dbclick_status));
+
+    m_pConfig->set(ConfigKey("[Library]","EditMetadata"),
+                    ConfigValue(checkBoxEditMetadata->checkState()));
+    m_pLibrary->setEditMedatata(checkBoxEditMetadata->checkState());
 
     QFont font = m_pLibrary->getTrackTableFont();
     if (m_originalTrackTableFont != font) {
@@ -331,13 +331,13 @@ void DlgPrefLibrary::slotApply() {
 }
 
 void DlgPrefLibrary::slotRowHeightValueChanged(int height) {
-    emit(setTrackTableRowHeight(height));
+    m_pLibrary->setRowHeight(height);
 }
 
 void DlgPrefLibrary::setLibraryFont(const QFont& font) {
     libraryFont->setText(QString("%1 %2 %3pt").arg(
         font.family(), font.styleName(), QString::number(font.pointSizeF())));
-    emit(setTrackTableFont(font));
+    m_pLibrary->setFont(font);
 
     // Don't let the row height exceed the library height.
     QFontMetrics metrics(font);

--- a/src/preferences/dialog/dlgpreflibrary.h
+++ b/src/preferences/dialog/dlgpreflibrary.h
@@ -20,8 +20,7 @@ class DlgPrefLibrary : public DlgPreferencePage, public Ui::DlgPrefLibraryDlg  {
     enum TrackDoubleClickAction {
         LOAD_TO_DECK,
         ADD_TO_AUTODJ_BOTTOM,
-        ADD_TO_AUTODJ_TOP,
-        EDIT_METADATA
+        ADD_TO_AUTODJ_TOP
     };
 
     DlgPrefLibrary(
@@ -51,8 +50,6 @@ class DlgPrefLibrary : public DlgPreferencePage, public Ui::DlgPrefLibraryDlg  {
     void requestAddDir(QString dir);
     void requestRemoveDir(QString dir, Library::RemovalType removalType);
     void requestRelocateDir(QString currentDir, QString newDir);
-    void setTrackTableFont(const QFont& font);
-    void setTrackTableRowHeight(int rowHeight);
 
   private slots:
     void slotRowHeightValueChanged(int);

--- a/src/preferences/dialog/dlgpreflibrarydlg.ui
+++ b/src/preferences/dialog/dlgpreflibrarydlg.ui
@@ -249,14 +249,14 @@
         </property>
        </widget>
       </item>
-      <item row="4" column="0">
-       <widget class="QCheckBox" name="checkBoxEditMetadata">
+      <item row="4" column="0" colspan="2">
+       <widget class="QCheckBox" name="checkBoxEditMetadataSelectedClicked">
         <property name="toolTip">
          <string>This enables instant metadata editing within the track table.
 Editing starts when clicking on an already selected item.</string>
         </property>
         <property name="text">
-         <string>Edit metadata</string>
+         <string>Edit metadata after clicking selected track</string>
         </property>
        </widget>
       </item>

--- a/src/preferences/dialog/dlgpreflibrarydlg.ui
+++ b/src/preferences/dialog/dlgpreflibrarydlg.ui
@@ -251,6 +251,10 @@
       </item>
       <item row="4" column="0">
        <widget class="QCheckBox" name="checkBoxEditMetadata">
+        <property name="toolTip">
+         <string>This enables instant metadata editing within the track table.
+Editing starts when clicking on an already selected item.</string>
+        </property>
         <property name="text">
          <string>Edit metadata</string>
         </property>

--- a/src/widget/wlibrarytableview.cpp
+++ b/src/widget/wlibrarytableview.cpp
@@ -18,10 +18,12 @@ WLibraryTableView::WLibraryTableView(QWidget* parent,
           m_pConfig(pConfig),
           m_vScrollBarPosKey(vScrollBarPosKey) {
 
+    loadVScrollBarPosState();
+
     // Setup properties for table
 
     // Editing starts when clicking on an already selected item.
-    setEditTriggers(QAbstractItemView::SelectedClicked);
+    setEditTriggers(QAbstractItemView::SelectedClicked|QAbstractItemView::EditKeyPressed);
 
     //Enable selection by rows and extended selection (ctrl/shift click)
     setSelectionBehavior(QAbstractItemView::SelectRows);
@@ -134,4 +136,12 @@ void WLibraryTableView::setTrackTableRowHeight(int rowHeight) {
     int fontHeightPx = metrics.height();
     verticalHeader()->setDefaultSectionSize(math_max(
                                                 rowHeight, fontHeightPx));
+}
+
+void WLibraryTableView::setSelectedClick(bool enable) {
+    if (enable) {
+        setEditTriggers(QAbstractItemView::SelectedClicked|QAbstractItemView::EditKeyPressed);
+    } else {
+        setEditTriggers(QAbstractItemView::EditKeyPressed);
+    }
 }

--- a/src/widget/wlibrarytableview.h
+++ b/src/widget/wlibrarytableview.h
@@ -51,6 +51,7 @@ class WLibraryTableView : public QTableView, public virtual LibraryView {
     void restoreVScrollBarPos();
     void setTrackTableFont(const QFont& font);
     void setTrackTableRowHeight(int rowHeight);
+    void setSelectedClick(bool enable);
 
   private:
     void loadVScrollBarPosState();

--- a/src/widget/wtracktableview.cpp
+++ b/src/widget/wtracktableview.cpp
@@ -105,8 +105,6 @@ WTrackTableView::WTrackTableView(QWidget * parent,
     connect(m_pCoverMenu, SIGNAL(reloadCoverArt()),
             this, SLOT(slotReloadCoverArt()));
 
-    setEditTriggers(QAbstractItemView::EditKeyPressed);
-
     // Create all the context m_pMenu->actions (stuff that shows up when you
     //right-click)
     createActions();
@@ -563,8 +561,6 @@ void WTrackTableView::slotMouseDoubleClicked(const QModelIndex &index) {
     } else if (doubleClickAction == DlgPrefLibrary::ADD_TO_AUTODJ_TOP
         && modelHasCapabilities(TrackModel::TRACKMODELCAPS_ADDTOAUTODJ)) {
         sendToAutoDJ(PlaylistDAO::AutoDJSendLoc::TOP);
-    } else if (doubleClickAction == DlgPrefLibrary::EDIT_METADATA) {
-        edit(index);
     }
 }
 

